scripts/kube-init.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+set -x
+
+function clean_exit(){
+    local error_code="$?"
+    local spawned=$(jobs -p)
+    if [ -n "$spawned" ]; then
+        kill $(jobs -p)
+    fi
+    return $error_code
+}
+
+trap "clean_exit" EXIT
+
+# Switch off SE-Linux
+setenforce 0
+
+# Install docker if needed
+path_to_executable=$(which docker)
+if [ -x "$path_to_executable" ] ; then
+    echo "Found Docker installation"
+else
+    curl -sSL https://get.docker.io | sudo bash
+fi
+docker --version
+
+# Get the latest stable version of kubernetes
+export K8S_VERSION=$(curl -sS https://storage.googleapis.com/kubernetes-release/release/stable.txt)
+echo "K8S_VERSION : ${K8S_VERSION}"
+
+echo "Starting docker service"
+sudo systemctl enable docker.service
+sudo systemctl start docker.service --ignore-dependencies
+echo "Checking docker service"
+sudo docker ps
+
+# Run the docker containers for kubernetes
+echo "Starting Kubernetes containers"
+sudo docker run \
+    --volume=/:/rootfs:ro \
+    --volume=/sys:/sys:ro \
+    --volume=/var/lib/docker/:/var/lib/docker:rw \
+    --volume=/var/lib/kubelet/:/var/lib/kubelet:rw \
+    --volume=/var/run:/var/run:rw \
+    --net=host \
+    --pid=host \
+    --privileged=true \
+    --name=kubelet \
+    -d \
+    gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} \
+    /hyperkube kubelet \
+        --containerized \
+        --hostname-override="127.0.0.1" \
+        --address="0.0.0.0" \
+        --api-servers=http://localhost:8080 \
+        --config=/etc/kubernetes/manifests \
+        --allow-privileged=true --v=2
+
+
+echo "Download Kubernetes CLI"
```

Comment:
we can first check if kubectl is available. if you want to always download an stable version, would suggest to download it in a temp folder and remove it at the end.